src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1391,6 +1697,218 @@ public override DataTable GetSchemaTable()
+                if (lastColumnLen != -1)
+                    Buffer.ReadPosition += lastColumnLen;
+                var len = Buffer.ReadInt32();
+                _columns.Add((Buffer.ReadPosition, len));
+            }
+
+            (Buffer.ReadPosition, ColumnLen) = _columns[column];
+            _column = column;
+            PosInColumn = 0;
+        }
+
+        /// <summary>
+        /// Seeks to the given column. The 4-byte length is read and stored in <see cref="ColumnLen"/>.
+        /// </summary>
+        async Task SeekToColumnSequential(int column, bool async)
+        {
+            if (_column == -1)
+            {
+                // Starting a new row.
+
+                // We currently assume that the row's number of columns is identical to the description's
+                await Buffer.Ensure(2, async);
+#if DEBUG
+                _numColumns = Buffer.ReadInt16();
+                Debug.Assert(RowDescription.NumFields == _numColumns);
+#else
+                Buffer.ReadPosition += 2;
+#endif
+            }
+
+            if (column < 0 || column >= _numColumns)
+                throw new IndexOutOfRangeException("Column index out of range");
+
+            if (column < _column)
+                throw new InvalidOperationException($"Invalid attempt to read from column ordinal '{column}'. With CommandBehavior.SequentialAccess, you may only read from column ordinal '{_column}' or greater.");
+
+            if (column == _column)
+                return;
+
+            // Need to seek forward
+
+            // Shut down any streaming going on on the column
+            if (ColumnStream != null)
+            {
+                ColumnStream.Dispose();
+                ColumnStream = null;
+                // Disposing the stream leaves us at the end of the column
+                PosInColumn = ColumnLen;
+            }
+
+            // Skip to end of column if needed
+            // TODO: Simplify by better initializing _columnLen/_posInColumn
+            var remainingInColumn = ColumnLen == -1 ? 0 : ColumnLen - PosInColumn;
+            if (remainingInColumn > 0)
+                await Buffer.Skip(remainingInColumn, async);
+
+            // Skip over unwanted fields
+            for (; _column < column - 1; _column++)
+            {
+                await Buffer.Ensure(4, async);
+                var len = Buffer.ReadInt32();
+                if (len != -1)
+                    await Buffer.Skip(len, async);
+            }
+
+            await Buffer.Ensure(4, async);
+            ColumnLen = Buffer.ReadInt32();
+            PosInColumn = 0;
+            _column = column;
+        }
+
+        Task SeekInColumn(int posInColumn, bool async)
+        {
+            if (_isSequential)
+                return SeekInColumnSequential(posInColumn, async);
+
+            if (posInColumn > ColumnLen)
+                posInColumn = ColumnLen;
+            Buffer.ReadPosition = _columns[_column].Offset + posInColumn;
```

Comment:
Add an empty line after the conditional block.

Reply:
OK